lib/View/welcome_screen.dart

```diff
@@ -0,0 +1,149 @@
+              const Spacer(),
+              Image.asset(
+                'assets/images/streamicon.png',
+                width: 150.0,
+                height: 150.0,
+              ),
+              const SizedBox(height: 24),
+              const Text(
+                'Welcome to Gocast',
+                textAlign: TextAlign.center,
+                style: TextStyle(
+                  fontSize: 24,
+                  fontWeight: FontWeight.bold,
+                  color: Colors.black,
+                ),
+              ),
+              const SizedBox(height: 8),
+              const Text(
+                "Here's a good place for a brief overview of the app or its key features.",
+                textAlign: TextAlign.center,
+                style: TextStyle(
+                  fontSize: 16,
+                  color: Colors.black54,
+                ),
+              ),
+              const SizedBox(height: 48),
+              Row(
+                mainAxisSize: MainAxisSize.min,
+                mainAxisAlignment: MainAxisAlignment.center,
+                children: [
+                  _indicatorDot(true),
+                  _indicatorDot(false),
+                  _indicatorDot(false),
+                ],
+              ),
+              const Spacer(),
+              ElevatedButton(
+                style: ElevatedButton.styleFrom(
+                  foregroundColor: Colors.white,
+                  backgroundColor: Colors.blue[900],
+                  padding: const EdgeInsets.symmetric(vertical: 16.0),
+                  shape: RoundedRectangleBorder(
+                    borderRadius: BorderRadius.circular(15.0),
+                  ),
+                ),
+                child: const Text(
+                  'TUM Login',
+                  style: TextStyle(fontSize: 18),
+                ),
+                onPressed: () => handleSSOLogin(
+                  context,
+                  ref,
+                  usernameController,
+                  passwordController,
+                ),
+              ),
+              const SizedBox(height: 12),
+              OutlinedButton(
+                style: OutlinedButton.styleFrom(
+                  side: const BorderSide(color: Colors.blue),
+                  foregroundColor: Colors.blue[900],
+                  padding: const EdgeInsets.symmetric(vertical: 16.0),
+                  shape: RoundedRectangleBorder(
+                    borderRadius: BorderRadius.circular(15.0),
+                  ),
+                ),
+                child: const Text(
+                  'Continue without',
+                  style: TextStyle(fontSize: 18),
+                ),
+                onPressed: () {},
+              ),
+              const SizedBox(height: 12),
+              InkWell(
+                onTap: () {
+                  Navigator.push(
+                    context,
+                    MaterialPageRoute(
+                      builder: (context) => const InternalloginScreen(),
```

Comment:
use named routes, as defined in the main app configuration (MaterialApp)

Reply:
I think this should be done in second PR.

Reply:
same for this.